### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by Maciej3206**
nwdNww: final conditions (for each function) - is this the optimal way to ensuring the positive value?

### CODE REVIEW

File: module2/homework/addEven/addEven.hpp [hunk 1]
```diff
@@ -2,7 +2,9 @@
 #include <vector>
 
 int addEven(const std::vector<int>& numbers) {
-    // TODO: Your implementation goes here
-    // Below return is only to make this function compile now
-    return -1;
+    int suma=0;
```
Comment (Kasperice) [inline module2/homework/addEven/addEven.hpp:5]: Try to use english names for variables

Comment (Maciej3206) [reply module2/homework/addEven/addEven.hpp:5]: Thank You, I will look after that.

File: module2/homework/generateSequence/generateSequence.hpp [hunk 2]
```diff
@@ -2,6 +2,12 @@
 #include <vector>
 
 std::vector<int> generateSequence(int count, int step) {
-    // TODO: Implement me :)
-    return {};
-}
+    if (count<=0) {return {};}
+    else {
+        std::vector<int> vec {step};
+        for (int i=1; i<count; i++){
```
Comment (piezol) [inline module2/homework/generateSequence/generateSequence.hpp:8]:
```suggestion
        for (int i = 1; i < count; i++){
```
Using spaces next to operators improve readability

File: module2/homework/generateSequence/generateSequence.hpp [hunk 3]
```diff
@@ -2,6 +2,12 @@
...
+            vec.push_back(step * (1 + i));
+        }
+     return vec;
```
Comment (piezol) [inline module2/homework/generateSequence/generateSequence.hpp:11]:
```suggestion
        return vec;
```
That return statement is within 'else' block and should be on the same indentation level as the rest of the code in that block.

File: module2/homework/nwdNww/nwdNww.hpp [hunk 4]
```diff
@@ -1,11 +1,24 @@
...
+    return (lhs>0) ? lhs : lhs * (-1) ;
 }
-
 int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
-}
+if (lhs==0 || rhs==0) {return 0;}
+long long mul=lhs*rhs;
+   unsigned int t;
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:16]:
```suggestion
    if (lhs==0 || rhs==0) {
        return 0;
    }
    long long mul=lhs*rhs;
    unsigned int t;
```
Indendation. Please avoid one-liner if-returns.

File: module2/homework/addEven/addEven.hpp (hunk 1 above)
Comment (nauka-programowania-MA) [inline module2/homework/addEven/addEven.hpp:5]: use english names and spaces :)

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 5]
```diff
@@ -3,6 +3,10 @@
...
-    // TODO: Implement me :)
-    return {};
-}
+    int max=vec[0];
+    for (const auto &it : vec)
+    {
+        if (it>max) {max=it;}
+    }
+return max;
+}
```
Comment (nauka-programowania-MA) [inline module2/homework/maxOfVector/maxOfVector.hpp:12]: new line at end of file

File: module2/homework/nwdNww/nwdNww.hpp [hunk 6]
```diff
@@ -1,11 +1,24 @@
 #pragma once
 
 int NWD(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+    unsigned int t;
+    while (rhs)
+    {
+        t = rhs;
```
Comment (nauka-programowania-MA) [inline module2/homework/nwdNww/nwdNww.hpp:7]: tmp, or temporary will be better name for `t`

File: module2/homework/nwdNww/nwdNww.hpp (hunk 4 above)
Comment (nauka-programowania-MA) [inline module2/homework/nwdNww/nwdNww.hpp:16]: code format

File: module2/homework/nwdNww/nwdNww.hpp [hunk 7]
```diff
@@ -1,11 +1,24 @@
...
+long long mul=lhs*rhs;
+   unsigned int t;
+    while (rhs)
+    {
+        t = rhs ;
+        rhs = lhs % rhs;
+        lhs = t;
+    }
+    return (mul /lhs)>=0 ? (mul /lhs) : (mul /lhs) *(-1);
+}
```
Comment (nauka-programowania-MA) [inline module2/homework/nwdNww/nwdNww.hpp:24]: new line at end of file

**Comment by ziobron** [comment]
> nwdNww: final conditions (for each function) - is this the optimal way to ensuring the positive value?

Yes, it passes all tests :)

+4 XP addEven
+6 XP NwwNwd
+5 XP maxOfVector
+5 XP generateSequence
+8 XP delivery before 31.05

{ "requirement_ids": [1119, 1631, 1119, 1633, 1119, 1634, 1119, 1635], "points": [2, 4, 2, 6, 2, 5, 2, 5] }

**Comment by ziobron** [comment]
> nwdNww: final conditions (for each function) - is this the optimal way to ensuring the positive value?

But there is a function `std::abs` which returns an absolute value.

**Pull request closed without merging**